vpr/src/place/noc_place_utils.cpp

```diff
@@ -0,0 +1,643 @@
+
+    // go through the moved blocks and process them only if they are NoC routers
+    for (int iblk = 0; iblk < blocks_affected.num_moved_blocks; ++iblk) {
+        ClusterBlockId blk = blocks_affected.moved_blocks[iblk].block_num;
+
+        // check if the current moved block is a noc router
+        if (noc_traffic_flows_storage->check_if_cluster_block_has_traffic_flows(blk)) {
+            // current block is a router, so re-route all the traffic flows it is a part of //
+
+            // get all the associated traffic flows for the logical router cluster block
+            const std::vector<NocTrafficFlowId>* assoc_traffic_flows = noc_traffic_flows_storage->get_traffic_flows_associated_to_router_block(blk);
+
+            // now check if there are any associated traffic flows
+            if (assoc_traffic_flows != nullptr) {
+                // There are traffic flows associated to the current router block so process them
+                for (auto& traffic_flow_id : *assoc_traffic_flows) {
+                    // first check to see whether we have already reverted the current traffic flow and only revert it if we haven't already.
+                    if (reverted_traffic_flows.find(traffic_flow_id) == reverted_traffic_flows.end()) {
+                        // Revert the traffic flow route by re-routing it
+                        re_route_traffic_flow(traffic_flow_id, *noc_traffic_flows_storage, noc_ctx.noc_model, noc_ctx.noc_flows_router, place_ctx.block_locs);
+
+                        // make sure we do not revert this traffic flow again
+                        reverted_traffic_flows.insert((NocTrafficFlowId)traffic_flow_id);
+                    }
+                }
+            }
+        }
+    }
+
+    return;
+}
+
+void re_route_traffic_flow(NocTrafficFlowId traffic_flow_id, NocTrafficFlows& noc_traffic_flows_storage, NocStorage& noc_model, NocRouting* noc_flows_router, const vtr::vector_map<ClusterBlockId, t_block_loc>& placed_cluster_block_locations) {
+    // get the current traffic flow info
+    const t_noc_traffic_flow& curr_traffic_flow = noc_traffic_flows_storage.get_single_noc_traffic_flow((NocTrafficFlowId)traffic_flow_id);
+
+    /*  since the current traffic flow route will be 
+     * changed, first we need to reduce the bandwidh
+     * usage of all links that are part of
+     * the existing traffic flow route
+     */
+    const std::vector<NocLinkId>& curr_traffic_flow_route = noc_traffic_flows_storage.get_traffic_flow_route(traffic_flow_id);
+    update_traffic_flow_link_usage(curr_traffic_flow_route, noc_model, link_usage_update_state::decrement, curr_traffic_flow.traffic_flow_bandwidth);
+
+    // now get the re-routed traffic flow route and update all the link usages with this reverted route
+    std::vector<NocLinkId>& re_routed_traffic_flow_route = get_traffic_flow_route((NocTrafficFlowId)traffic_flow_id, noc_model, noc_traffic_flows_storage, noc_flows_router, placed_cluster_block_locations);
```

Comment:
I think the (NocTrafficFlowId) cast is unnecessary (traffic_flow_id already should have the right type).

Reply:
Added to an issue and will be updated later